### ISSUE TITLE
http://apisof.net/ is returning HTTP 500

### ISSUE THREAD

**Issue by ogwh** [dotnet/corefx#26652]

**Comment by joshfree** [comment]
Are you still seeing an issue @ogwh?

**Comment by ogwh** [comment]
Yep, the only thing I see on the page is the string "The page cannot be displayed because an internal server error has occurred."

Can confirm I'm still seeing a 500 response code also.

**Comment by joshfree** [comment]
It works for me in both edge and chrome.

I can repro the HTTP 500 in IE11.

**Comment by ogwh** [comment]
I've just checked Edge and it's loading there for me too but initially /very/ slowly... tried a private tab in Chrome and it works there. Looking at the HTTP headers literally the only difference is that in the private tab the "Cache-Control: max-age=0" header isn't sent. That's literally the only difference between the two requests. So perhaps a bug related to the handling of the header?

**Comment by ogwh** [comment]
Found it!

It has something to do with the ARRAffinity cookie. Mine is set to "2a48e0f6708a66f806e63c6f5f2684db14d155e35029664053ffa3d81e438d1e".

Once I replayed the request in Fiddler but deleted the cookie I got a 302 redirect to HTTPS. I don't know what data is associated with that cookie but whatever it is, it's causing the 500 error.

So the two requests weren't quite literally identical...

Earlier I got the same response (same generic error page with a 500) from notebooks.azure.com whenever I tried to run a Jupyter notebook with a "é" (unicode) char in the file name. ARRAffinity seems to be associated with Azure (I don't really use it)... perhaps there's a bug in an Azure component and it's actually that component erroring and not whichever servers are set behind it? Seems a bizarre hypothesis but... I'll clear my cookies and try to reproduce (suddenly I hate 2FA).

**Comment by ogwh** [comment]
I thought maybe it was being injected by a load balancer shared across Azure sites or something equally daft and that something about it was a consequence of the earlier Azure Notebooks 500 error issue (which is open on their GH project).

But yeah, if you set the value of the cookie to that given in my previous comment you get a 500, just reproduced by editing the cookie in Chrome after having cleared them all. Could not reproduce by triggering the Azure Notebooks error prior to visiting apisof.net.

I'm guessing your IE11 has a cookie that's also somehow triggering the same problem? Just looks like a hash to me, only hex digits so whatever data is triggering it is no doubt in yours or someone else's DB?

**Comment by terrajobst** [comment]
I wrote apisof.net. The sites work off an in-memory representation. Since I added the new dynamic search, it seems there are cases where the site simply runs out-of-memory. Given that Azure has multiple machines serving the side, I believe the browser differences as mostly a red herring; I suspect the root cause is the combination of headers & how the load balancer decides which server to send you to.

Ideally we'd get some real web devs to fix the site as I mostly wrote it to learn the new ASP.NET :-)
